tests/sqlparser_snowflake.rs

```diff
@@ -1064,3 +1064,40 @@ fn test_snowflake_trim() {
         snowflake().parse_sql_statements(error_sql).unwrap_err()
     );
 }
+
+#[test]
+fn test_snowflake_rank() {
+    let real_sql = "SELECT col_1, col_2, LAG(col_2) IGNORE NULLS OVER (ORDER BY col_1) FROM t1";
+    assert_eq!(snowflake().verified_stmt(real_sql).to_string(), real_sql);
+
+    let first_sql = "SELECT column1, column2, FIRST_VALUE(column2) OVER (PARTITION BY column1 ORDER BY column2 NULLS LAST) AS column2_first FROM t1";
+    assert_eq!(snowflake().verified_stmt(first_sql).to_string(), first_sql);
+
+    let sql_only_select = "SELECT LAG(col_2,1,0) OVER (PARTITION BY col_3 ORDER BY col_1)";
+    let select = snowflake().verified_only_select(sql_only_select);
```

Comment:
It seems this rather extends support for window function options Ignore/Respect NULLs vs snowflake specific, since the options are standard sql (e.g the test `first_sql` and `sql_select_only` should already be parsed without the MR)? If so it might be reasonable to instead extend the `parse_function()` to optionally accept an option where applicable like the following?
```
enum WindowFunctionOption {
	RespectNulls,
	IgnoreNulls
}
```
Since parse_function already contains most of the functionality being added, and prevents window functions from being parsed into potentially different types in the AST which might not be desirable.

Reply:
Thank you @iffyio !
I extracted the null clause to an enum, and moved it to `parse_function`.
I will appreciate any feedback.

Thanks!

Reply:
Oh nice!

One thing that stood out was the mismatch between the field name and the typename:
```rust
nulls_clause: Option<WindowFunctionOption>
```
was curious and took a quick look and it turns out the option is mostly called [null treatment clause](https://www.google.com/search?q=window+function+null+treatment+clause) - maybe we can rename accordingly and so they're consistent?

```
null_treatment: Option<NullTreatment>
```

Other than that I think the changes look good!

Reply:
Thanks @iffyio ! I looked for a better name but didn't found one, you did it! 
Renamed :)